src/EllCrv/Finite.jl

```diff
@@ -816,3 +817,132 @@ function trace_of_frobenius(E::EllCrv{T}, n::Int) where T<:FinFieldElem
   end
 end
 
+################################################################################
+#
+#  Supersingular Elliptic Curves
+#
+################################################################################
+
+
+#Following Identifying supersingular elliptic curves - Andrew V. Sutherland
+@doc Markdown.doc"""
+    is_supersingular(E::EllCrv{T}) where T <: FinFieldElem
+Return true when the elliptic curve is supersingular. The result is proven to be correct.
+"""
+funct
+function is_supersingular(E::EllCrv{T}) where T <: FinFieldElem
+  K = base_field(E)
+  
+  #This is still stupid. Can't embed GF(fmpz(p)) or (GF(fmpz(p,n))) into GF(fmpz(p), n)
+  p = Int(characteristic(K))
+  j = j_invariant(E)
+  
+  if j^(p^2) != j
+    return false
+  end
+  
+  if p<= 3
+    return j == 0
+  end
+  
+  L = GF(p, 2)
+  Lx, X = PolynomialRing(L, "X")
+  Lxy, Y = PolynomialRing(Lx, "Y")
+  Phi2 = X^3 + Y^3 - X^2*Y^2 + 1488*(X^2*Y + Y^2*X) - 162000*(X^2 + Y^2) + 40773375*X*Y + 8748000000*(X + Y) - 157464000000000
+  
+  j = L(j)
+  
+  js = roots(Phi2(j))
+  
+  if length(js) < 3
+    return false
+  end
+  
+  newjs = [j, j, j]
+  f = fq_nmod_poly[zero(Lx), zero(Lx), zero(Lx)]
+  
+  m = nbits(p) - 1
+  for k in (1 : m)
+    for i in (1 : 3)
+      f[i] = divexact(Phi2(js[i]), X - newjs[i])
+      newjs[i] = js[i]
+      froots = roots(f[i])
+      if isempty(froots)
+        return false
+      end
+      js[i] = froots[1]
+    end
+  end
+  return true
+end
+
+@doc Markdown.doc"""
+    is_ordinary(E::EllCrv{T}) where T <: FinFieldElem
+Return true when the elliptic curve is ordinary, i.e. not supersingular.
+"""
+function is_ordinary(E::EllCrv{T}) where T <: FinFieldElem
+  return !is_supersingular(E)
+end
+
+#Following Identifying supersingular elliptic curves - Andrew V. Sutherland
+@doc Markdown.doc"""
+    supersingular_monte_carlo(E::EllCrv{T}) where T <: FinFieldElem
+Uses a probabilistic algorithm to test whether E is supersingular or not.
+If the function returns false, the curve is proven to be ordinary.
+If the function returns true, there is a high chance the curve is supersingular,
+but the result hasn't been proven.
+"""
+function supersingular_monte_carlo(E::EllCrv{T}) where T <: FinFieldElem
```

Comment:
```suggestion
function is_probable_supersingular(E::EllCrv{T}) where T <: FinFieldElem
```
(To be consistent with `is_probable_prime`).

Reply:
Needs to be then adjusted in the tests.